lib/upipe-alsa/upipe_alsa_source.c

```diff
@@ -184,8 +187,9 @@ static struct upipe *upipe_alsource_alloc(struct upipe_mgr *mgr,
     upipe_alsource->rate = 48000;
     upipe_alsource->channels = 2;
     upipe_alsource->format = SND_PCM_FORMAT_FLOAT;
-    upipe_alsource->uri = strdup(DEFAULT_DEVICE);
+    upipe_alsource->uri = NULL;
```

Comment:
Is this an intended change ?

Reply:
Yes but it seems incomplete, will fix it

Reply:
Hm actually it seems good as-is.

upipe_alsource_open() is only called from set_uri, and set_uri needs uri to be set, effectively overwriting this first assignement.